src/shell/command_line.cpp

```diff
@@ -161,6 +161,23 @@ bool CommandLine::FindStringBegin(const char* const begin, std::string& value,
 	return false;
 }
 
+bool CommandLine::FindStringCaseInsensitiveBegin(const char* const begin,
+                                                 std::string& value,
+                                                 bool remove)
+{
+	size_t len = strlen(begin);
+	for (cmd_it it = cmds.begin(); it != cmds.end(); ++it) {
+		if (ciequals_n(begin, (*it).c_str(), len)) {
+			value = ((*it).c_str() + len);
+			if (remove) {
+				cmds.erase(it);
+			}
+			return true;
+		}
+	}
```

Comment:
I'm not a fan of introducing yet another C-style string comparison function. We should try hard not to do this and convert C strings to `std::string` instead.

Just convert the snippet between the begin/end markers to a `std::string`, then use `iequals` on it from `string_utils.h`. `ciequals_n` here is a `micro-optimisation.` It does not matter one bit if we create a few temporariy `std::strings` when performing this find.

Reply:
Actually, I'm pretty sure you can use string view with `iequals` because it's templated. So you won't even need to create temporary copies (not that it would matter).

Reply:
Reworked to use `iequals` based solution.